applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -918,8 +918,9 @@ static void disconnect_peripheral(struct hids_peripheral *per)
 			uint8_t report_id = bt_hogp_rep_id(rep);
 			size_t size = bt_hogp_rep_size(rep);
 
-			__ASSERT_NO_MSG((report_id != REPORT_ID_RESERVED) &&
```

Comment:
Can we always rely on the data when report ID is inside bonds that are defined here?  Maybe we could use `bt_hogp_assign_check` that would return false if discovery was not complete?

Reply:
Then we would not try to forward empty HID reports at all for a given peripheral

Reply:
will check the `bt_hogp_assign_check`. please approve so we merge this and I will start another PR with the other check.

Reply:
This check is already done before calling `disconnect_peripheral`

Reply:
I guess this was the reason why we had asserts there in the first place. the discovery was completed but discovery data was set to zero. This can be the problem as you have chain elements allocated before actually being filled.

Reply:
So both the report_id and report size are set to zeros? Maybe we should add a comment and use this as a condition (maybe that could be even done in `hogp_client` subsystem to prevent adding the workaround to the application).

Reply:
If the data is actually uninitialized we still may send HID report that has improper size or ID that is not supported by given peripheral

Reply:
Discussed offline

Reply:
Discussed offline. checking report id == 0 should be enough as data is now zeroed during the discovery process.